lib/voyager_helpers/queries.rb

```diff
@@ -251,29 +251,40 @@ def record_ids_for_barcode(item_barcode)
         )
       end
 
-      def all_recap_bib_ids
+      def location_bib_ids(locations)
+        locations = OCI8::in_cond(:locations, locations)
         %Q(
-          SELECT bib_master.bib_id
-          FROM (
-                 (
-                   (bib_master JOIN bib_mfhd ON bib_master.bib_id = bib_mfhd.bib_id) 
-                 JOIN mfhd_master ON bib_mfhd.mfhd_id = mfhd_master.mfhd_id) 
-               JOIN mfhd_item ON mfhd_master.mfhd_id = mfhd_item.mfhd_id) 
-          WHERE mfhd_master.location_id IN (#{recap_locations})
-          AND bib_master.suppress_in_opac = 'N'
-          AND mfhd_master.suppress_in_opac = 'N'
+          SELECT
+            bib_master.bib_id
+          FROM bib_master
+            JOIN bib_mfhd
+              ON bib_master.bib_id = bib_mfhd.bib_id
+            JOIN mfhd_master
+              ON bib_mfhd.mfhd_id = mfhd_master.mfhd_id
+            JOIN mfhd_item
+              ON mfhd_master.mfhd_id = mfhd_item.mfhd_id
+          WHERE
+            mfhd_master.location_id IN (#{locations.names})
+            AND bib_master.suppress_in_opac = 'N'
+            AND mfhd_master.suppress_in_opac = 'N'
           GROUP BY bib_master.bib_id
           ORDER BY bib_master.bib_id
         )
       end
 
-      def recap_mfhd_ids(bib_id)
+      def location_mfhd_ids(bib_ids, locations)
+        bib_ids = OCI8::in_cond(:bib_ids, bib_ids)
+        locations = OCI8::in_cond(:locations, locations)
         %Q(
-        SELECT bib_mfhd.mfhd_id
-        FROM bib_mfhd JOIN mfhd_master ON bib_mfhd.mfhd_id = mfhd_master.mfhd_id
-        WHERE (bib_id = #{bib_id}
-        and location_id IN (#{recap_locations})
-        and suppress_in_opac = 'N')
+        SELECT
+          bib_mfhd.mfhd_id
+        FROM bib_mfhd
+          JOIN mfhd_master
+            ON bib_mfhd.mfhd_id = mfhd_master.mfhd_id
+        WHERE
+          bib_id IN (#{bib_ids.names})
```

Comment:
I will be honest, I didn't know at the time how to bind a single value for a variable, as the in_cond method requires an array. I also thought at the time that it might be useful to be able to pass multiple bibs, but I think now that it muddles the functionality of the method. I have found documentation for binding single values instead of arrays, so I will work on limiting it to one bib id.